### PR TITLE
More system info

### PR DESCRIPTION
**Pull request by Liborsaf**

### CODE REVIEW

File: sonyapilib/device.py [hunk 1]
```diff
@@ -272,6 +281,32 @@ def _parse_ircc(self):
             self.ircc_url, method=HttpMethod.GET, raise_errors=True)
 
         upnp_device = f"{URN_UPNP_DEVICE}device"
+
+        self.friendly_name = self._find_device_info(response.text, "friendlyName",
```
Comment (alexmohr) [inline sonyapilib/device.py]: wouldn't it be easier to initalize the field right away instead of splitting decl and init?

File: sonyapilib/device.py [hunk 2]
```diff
@@ -320,6 +355,18 @@ def _parse_ircc(self):
 
             self._ircc_categories.add(category_info.text)
 
+    @staticmethod
+    def _find_device_info(text, info, upnp_device=None):
+        upnp_device = upnp_device or f"{URN_UPNP_DEVICE}device"
```
Comment (alexmohr) [inline sonyapilib/device.py:386]: can you add a test where upnp_device is None to test if the search is working then?

File: sonyapilib/device.py [hunk 3]
```diff
@@ -272,6 +281,32 @@ def _parse_ircc(self):
...
+                                                upnp_device=upnp_device)
+
+        icons = find_in_xml(
+            response.text,
+            [upnp_device,
+             f"{URN_UPNP_DEVICE}iconList",
+             (f"{URN_UPNP_DEVICE}icon", True),
+             f"{URN_UPNP_DEVICE}url"])
+
+        if not hasattr(self, 'ircc_base'):
```
Comment (alexmohr) [inline sonyapilib/device.py]: why check if the attrbute exists when it's always set in init?

Comment (Liborsaf) [reply sonyapilib/device.py]: `load_from_json` will not call `__init__` so it will not exist when loading old json files

File: sonyapilib/device.py (hunk 1 above)
Comment (Liborsaf) [reply sonyapilib/device.py]: We need `response.text` from ircc request, so I guess we can't move it to `__init__` ?

File: sonyapilib/device.py (hunk 2 above)
Comment (Liborsaf) [reply sonyapilib/device.py:386]: Will take a look at that in approximately an hour

File: sonyapilib/device.py (hunk 3 above)
Comment (alexmohr) [reply sonyapilib/device.py]: makes sense, I see there is a see there is a separate commit to address this, so I assume you discovered this when testing the integration? It probably would make sense to add a test case with the 'old' and new json device?

File: sonyapilib/device.py (hunk 1 above)
Comment (alexmohr) [reply sonyapilib/device.py]: sorry I missed that this is not in init.

File: sonyapilib/device.py (hunk 3 above)
Comment (Liborsaf) [reply sonyapilib/device.py]: Yeah, it didn't like the old config when I was testing it :D

I will add those tests in a few minutes.

Comment (Liborsaf) [reply sonyapilib/device.py]: Done

File: sonyapilib/device.py (hunk 2 above)
Comment (Liborsaf) [reply sonyapilib/device.py:386]: Done

**Comment by Liborsaf** [comment]
Added those tests, hopefully they are somehow good 🤔

File: tests/data/old.json [hunk 4]
```diff
@@ -0,0 +1,663 @@
+{
```
Comment (alexmohr) [inline tests/data/old.json]: can this be named `v0.5.0.json` and the new.json `0.6.0.json` (because next release will be 0.6.0)? So we can add more json files when adding more fields in new versions and easily extend the tests

File: tests/device_test.py [hunk 5]
```diff
@@ -279,6 +279,22 @@ def test_update_service_urls_error_response(self):
         device = self.create_device()
         device._update_service_urls()
 
+    @mock.patch('requests.get', side_effect=mocked_requests_get)
+    @mock.patch('requests.post', side_effect=mocked_requests_post)
+    def test_load_old_json_file(self, mocked_requests_post, mocked_requests_get):
```
Comment (alexmohr) [inline tests/device_test.py]: also here should be 0_5_0 instead of old

File: tests/device_test.py [hunk 6]
```diff
@@ -279,6 +279,22 @@ def test_update_service_urls_error_response(self):
...
+        content = read_file("data/old.json")
+        device = SonyDevice.load_from_json(content)
+
+        self.verify_json_load_fields(device)
+
+    @mock.patch('requests.get', side_effect=mocked_requests_get)
+    @mock.patch('requests.post', side_effect=mocked_requests_post)
+    def test_load_new_json_file(self, mocked_requests_post, mocked_requests_get):
```
Comment (alexmohr) [inline tests/device_test.py]: and 0_6_0 instead of new

**Comment by Liborsaf** [comment]
It seems that dmr contains even better system info, I have an idea, will try to implement one thing before merge.

File: tests/device_test.py (hunk 6 above)
Comment (Liborsaf) [reply tests/device_test.py]: Done

File: tests/device_test.py (hunk 5 above)
Comment (Liborsaf) [reply tests/device_test.py]: Done

File: tests/data/old.json (hunk 4 above)
Comment (Liborsaf) [reply tests/data/old.json]: Done

**Comment by Liborsaf** [comment]
In future I will try to make those tests (with dmr and only ircc) little better, but for now it seems to be ok and should work

**Comment by alexmohr** [comment]
Please squash commits and then it's ready to be merged.
